### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other fixes, this commit ensures the generated HTML
documentation for HexDocs.pm will become the main documentation
source for this Elixir library.

List of changes:
* Update gitignore
* Update copyright year
* Set and use latest ex_doc
* Set readme as main HTML doc
* Fix markdown
* Badges and more badges!
* Fix license link
* Add source reference by tagged version

Screenshot:

Screenshot-20210217222033-2740x1462

### CODE REVIEW

**Review by szymon-jez — APPROVED** [review]
Looking good

File: mix.exs [hunk 1]
```diff
@@ -28,10 +28,10 @@ defmodule ExPhoneNumber.Mixfile do
   defp deps do
     [
       {:sweet_xml, "~> 0.6.5"},
-      {:ex_doc, "~> 0.18.0", only: :dev, runtime: false},
+      {:credo, "~> 1.0.0", only: [:dev, :test], runtime: false},
+      {:ex_doc, ">= 0.0.0", only: :dev, runtime: false},
       {:ex_spec, "~> 2.0", only: :test},
-      {:excoveralls, "~> 0.10", only: :test},
-      {:credo, "~> 1.0.0", only: [:dev, :test], runtime: false}
+      {:excoveralls, "~> 0.10", only: :test}
```
Comment (szymon-jez) [inline mix.exs:34]: The changes look like done to order the list alphabetically, but they are not as `sweet_xml` is out of order. Either let's put `sweet_xml` last to have alphabetical order or revert this change.

**Review by tomekowal — APPROVED** [review]

**Comment by tomekowal** [comment]
Thank you @kianmeng for your input :heart:

**Comment by szymon-jez** [comment]
@kianmeng Thank you very much. Merged.
